src/js/modules/rpc/service.ts

```diff
@@ -13,62 +13,80 @@ import { safeLoadAll } from "js-yaml";
 import { join, resolve } from "path";
 import * as fs from "fs";
 import { promisify } from "util";
-import { newLogger } from "../utilities/Logging";
+import { createLoggerFromLogFilePath } from "../utilities/Logging";
+import { Logger } from "winston";
 
-const logger = newLogger("service");
+class Service {
+  private logger: Logger;
 
-// read yaml config file
-const readConfigFile = (confPath: string): Promise<Record<string, any>> => {
-  try {
-    logger.info(`Reading from config file: ${confPath}`);
-    return fs.promises.readFile(confPath).then((file) => safeLoadAll(file)[0]);
-  } catch (e) {
-    return Promise.reject(new Error(`Error reading config file: ${e.message}`));
-  }
-};
+  constructor() {
+    const configPathArg = process.argv.splice(2)[0];
+    const defaultConfigPath = "/var/lib/redpanda/conf/redpanda.yaml";
+    const defaultCoprocessorPath = "/var/lib/redpanda/coprocessor";
+    const defaultLogFilePath = "/var/lib/redpanda/coprocessor/logs/wasm";
+    const configPath = configPathArg
+      ? resolve(configPathArg)
+      : defaultConfigPath;
 
-/*
-  validate scaffolding on coprocessor path, if the expected folders don't
-  exist, it'll create them
-*/
-const validateOrCreateScaffolding = (directoryPath: string): Promise<void> => {
-  const exists = promisify(fs.exists);
-  const validations = ["active", "inactive", "submit"].map((folder) => {
-    const path = join(directoryPath, folder);
-    return exists(path).then((exist) => {
-      if (!exist) {
-        logger.info(`Creating part of scaffolding ${path}`);
-        return fs.promises.mkdir(path, { recursive: true });
-      }
-    });
-  });
-  return Promise.all(validations).then(() => null);
-};
+    this.readConfigFile(configPath).then((config) => {
+      const port = config?.redpanda?.coproc_supervisor_server || 43189;
+      const path = config?.coproc_engine?.path || defaultCoprocessorPath;
+      const logFilePath =
+        config?.coproc_engine?.logFilePath || defaultLogFilePath;
+      // creating logger
+      const createCoprocLogger = createLoggerFromLogFilePath(logFilePath);
+      this.logger = createCoprocLogger("service");
 
-function main() {
-  // read config file path argument
-  logger.info("Starting redpanda wasm service...");
-  const configPathArg = process.argv.splice(2)[0];
-  const defaultConfigPath = "/var/lib/redpanda/conf/redpanda.yaml";
-  const defaultCoprocessorPath = "/var/lib/redpanda/coprocessor";
-  // resolve config path or assign default value
-  const configPath = configPathArg ? resolve(configPathArg) : defaultConfigPath;
-  logger.info(`Using redpanda configuration path: ${configPath}`);
+      this.logger.info("Starting redpanda wasm service...");
+      this.logger.info(`Using redpanda configuration path: ${configPath}`);
+      this.logger.info(`Using root scaffolding path: ${path}`);
+      this.logger.info(`Reading from config file: ${configPath}`);
```

Comment:
This is redundant since this is already logged: https://github.com/vectorizedio/redpanda/pull/303/commits/bd916e4f7f3c613de0d19aedbb35aaf1cc565d41#diff-d0f577ba35c7941840e3a4525d319dd646654090124810dd1a909e9cf874fc71R41

Reply:
I see, I'll remove